lib/fs.js

```diff
@@ -688,14 +688,19 @@ fs.renameSync = function(oldPath, newPath) {
 };
 
 fs.truncate = function(path, len, callback) {
-  if (typeof path === 'number') {
+  if (Number.isFinite(path) && path >= 0)
     return fs.ftruncate(path, len, callback);
-  }
+
+  if (typeof path !== 'string')
+    throw new TypeError('path must be a string');
+
   if (typeof len === 'function') {
     callback = len;
     len = 0;
-  } else if (len === undefined) {
+  } else if (len === undefined || len === null) {
     len = 0;
+  } else if (!Number.isFinite(len) || len < 0) {
+    throw new TypeError('length must be a positive integer');
```

Comment:
@targos Man, that's wonderful. I didn't know about that function and that is exactly what we need here. Thanks :-) Updated the PR.